### PR TITLE
configure 300 second timeout to prevent CI hanging

### PR DESCRIPTION
**Pull request by graingert**
*Note*: Before submitting this pull request, please review our [contributing
guidelines](https://docs.celeryproject.org/en/master/contributing.html).

### CODE REVIEW

**Comment by graingert** [comment]
I don't think https://github.com/celery/celery/pull/6255 actually did anything as when `@pytest.mark.flaky` restarts a test it prints the message in the console so should not trigger travis killing the build

**Comment by graingert** [comment]
@auvipy those are lint errors on the default branch

File: .travis.yml [hunk 1]
```diff
@@ -10,9 +10,9 @@ python:
 os:
     - linux
 stages:
+  - lint
```
Comment (auvipy) [inline .travis.yml:13]: thanks but can you revert this?

Comment (graingert) [reply .travis.yml]: @auvipy I've done that, why though?

Comment (auvipy) [reply .travis.yml]: we used to run lint first for many years. but later we find out that, many initial contributors/comparatively irregular contributors do not get the benefit of running their code through the unit & integration test for lint errors and the don't get the benefit of early feedback loop on their actual code changes. so we come to the conclusion that we should let the unit and integration tests first.

**Pull request merged**